packages/web/src/executors/rollup/lib/update-package-json.ts

```diff
@@ -0,0 +1,63 @@
+import { relative } from 'path';
+import { ExecutorContext } from 'nx/src/config/misc-interfaces';
+import { ProjectGraphProjectNode } from 'nx/src/config/project-graph';
+import {
+  DependentBuildableProjectNode,
+  updateBuildableProjectPackageJsonDependencies,
+} from '@nrwl/workspace/src/utilities/buildable-libs-utils';
+import { writeJsonFile } from 'nx/src/utils/fileutils';
+import { PackageJson } from 'nx/src/utils/package-json';
+import { NormalizedWebRollupOptions } from './normalize';
+
+export function updatePackageJson(
+  options: NormalizedWebRollupOptions,
+  context: ExecutorContext,
+  target: ProjectGraphProjectNode,
+  dependencies: DependentBuildableProjectNode[],
+  packageJson: PackageJson
+) {
+  const type = options.format.includes('esm') ? 'module' : 'commonjs';
+  const exports = {
+    '.': {},
+  };
+
+  if (type === 'module') {
+    // `module` field is used by bundlers like rollup and webpack to detect ESM.
+    // May not be required in the future if type is already "module".
+    packageJson.module = './index.js';
+    packageJson.main = './index.js';
+  } else {
+    // Only if ESM is not used at all.
+    packageJson.main = './index.cjs';
+  }
+
+  if (options.format.includes('esm')) {
+    exports['.']['import'] = './index.js';
+  }
+  if (options.format.includes('umd') || options.format.includes('cjs')) {
+    exports['.']['require'] = './index.cjs';
+  }
```

Comment:
suggestion: we can also add `'types'` to the `exports['.']`. `typings` and `types` on `package.json` are synonymous but not sure if the same is applied to `exports` field.